cli/command/utils.go

```diff
@@ -115,5 +117,23 @@ func PruneFilters(dockerCli Cli, pruneFilters filters.Args) filters.Args {
 		pruneFilters.Add(parts[0], parts[1])
 	}
 
+	// Server APIs older than 1.29 will just ignore label filters.
+	// It's dangerous because a command like:
+	// `docker container prune --filter label=foo=bar`
+	// would just remove all stopped containers without considering the label
+	// The remote API should reject requests using unknown filters.
+	if pruneFilters.Get("label") != nil {
+		serverVersion, err := dockerCli.Client().ServerVersion(context.Background())
```

Comment:
Yup, https://github.com/docker/cli/blob/0a613971b943e8a0713e75884fb24e2aa18b0ff2/cli/command/cli.go#L200-L203 sets the client version to the server version if the server version is lower.